pinot-query-runtime/src/main/java/org/apache/pinot/query/service/QueryDispatcher.java

```diff
@@ -65,15 +67,16 @@ public QueryDispatcher() {
   public ResultTable submitAndReduce(long requestId, QueryPlan queryPlan,
       MailboxService<TransferableBlock> mailboxService, long timeoutMs, Map<String, String> queryOptions)
       throws Exception {
+    long deadlineNanos = System.nanoTime() + TimeUnit.MILLISECONDS.toNanos(timeoutMs);
```

Comment:
The timeOut passed here is after deducting whatever was spent in building QueryPlan right ?

Reply:
It is actually after plan. It is a good point. I moved the counting of start time to the place where we receive the request. So it is now counting all of the time including parsing and planning.